openhands/server/github.py

```diff
@@ -0,0 +1,72 @@
+import os
+
+import httpx
+
+from openhands.core.logger import openhands_logger as logger
+
+GITHUB_CLIENT_ID = os.getenv('GITHUB_CLIENT_ID', '').strip()
+GITHUB_CLIENT_SECRET = os.getenv('GITHUB_CLIENT_SECRET', '').strip()
+GITHUB_USER_LIST = None
+
+
+def load_github_user_list():
+    global GITHUB_USER_LIST
+    waitlist = os.getenv('GITHUB_USER_LIST_FILE')
+    if waitlist:
+        with open(waitlist, 'r') as f:
+            GITHUB_USER_LIST = [line.strip() for line in f if line.strip()]
+
+
+load_github_user_list()
+
+
+async def authenticate_github_user(auth_token) -> bool:
+    logger.info('Checking GitHub token')
+    if not GITHUB_USER_LIST:
+        return True
+
+    if not auth_token:
+        logger.warning('No GitHub token provided')
+        return False
+
+    login, error = await get_github_user(auth_token)
```

Comment:
"My code says Python, but my heart says Go" 😆

Reply:
😂 this was openhands!